site/content/api/Credentials-API.md

```diff
@@ -0,0 +1,327 @@
+
+**Message Flow**
+
+*TODO* add sequence diagram
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a *remove credentials* message in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name             | Mandatory | Location                 | Type           | Description |
+| :--------------- | :-------: | :----------------------- | :------------- | :---------- |
+| *subject*        | yes       | *properties*             | UTF-8 *string* | MUST contain the value `remove`. |
+
+The body of the message MUST consist of a single *AMQP Value* section containing a UTF-8 encoded string representation of a single JSON object as follows:
+
+~~~json
+{
+  "device-id": "${device_id}",
+  "type": "${credential_type}",
+  "auth-id": "${authentication_identifier}"
+}
+~~~
+
+The `device-id` property MUST contain the ID of the device from which the credentials should be removed.
+The `type` property indicates the type of credentials to remove. If set to `*` then all credentials of the device are removed (`auth-id` is ignored), otherwise only the credentials matching the `type` and `auth-id` are removed.
+The `auth-id` property is optional and if omitted, then all credentials of the specified `type` are removed.
+
+**Response Message Format**
+
+A response to a *remove credentials* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's body is empty.
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *204* | No Content, the credentials have been removed successfully. |
+| *404* | Not Found, there are no credentials registered matching the critera given in the request payload. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+# Standard Message Properties
+
+Due to the nature of the request/response message pattern of the operations of the Credentials API, there are some standard properties shared by all of the request and response messages exchanged as part of the operations.
+
+## Standard Request Properties
+
+The following table provides an overview of the properties shared by all request messages regardless of the particular operation being invoked.
+
+| Name             | Mandatory | Location                 | Type           | Description |
+| :--------------- | :-------: | :----------------------- | :------------- | :---------- |
+| *action*         | yes       | *application-properties* | UTF-8 *string* | MUST be set to the value defined by the particular operation being invoked. |
```

Comment:
I see "action" here instead of "subject" as already specified in the operation sections.